### PR TITLE
Add changes to handle baggage in jaeger propagator

### PR DESCRIPTION
**Pull request by akashsharma95**

### CODE REVIEW

File: src/Trace/Propagator/JaegerPropagator.php [hunk 1]
```diff
@@ -34,6 +34,8 @@ class JaegerPropagator implements PropagatorInterface
 
     const CONTEXT_HEADER_FORMAT = '%032s:%016s:%016s:%x';    //traceId, spanId are stored as hex strings in opencensus
 
+    const BAGGAGE_HEADER_PREFIX = 'rzpctx-';
```
Comment (asldevi) [inline src/Trace/Propagator/JaegerPropagator.php:37]: keep this configurable from the JaegerPropagator constructor, than hardcoding it in the library

File: src/Trace/Propagator/JaegerPropagator.php [hunk 2]
```diff
@@ -99,9 +105,33 @@ public function inject(SpanContext $context, HeaderSetter $setter)
 
         $value = sprintf(self::CONTEXT_HEADER_FORMAT, $traceId, $spanId, $parentID, $enabled);
 
+        // set baggage header
+        if (count($context->baggage()) > 0) {
+            foreach ($context->baggage() as $k => $v) {
+                $setter->set(strtolower(self::BAGGAGE_HEADER_PREFIX . $k), $v);
+            }
+        }
+
```
Comment (asldevi) [inline src/Trace/Propagator/JaegerPropagator.php:114]: make this a separate method

**Review by subintp — APPROVED** [review]

File: ext/opencensus_trace.c [hunk 3]
```diff
@@ -64,6 +64,8 @@ void opencensus_trace_rinit()
 
     OPENCENSUS_G(current_span) = NULL;
     OPENCENSUS_G(trace_id) = NULL;
+    ALLOC_HASHTABLE(OPENCENSUS_G(baggage));
+    zend_hash_init(OPENCENSUS_G(baggage), 16, NULL, NULL, 0);
```
Comment (subintp) [inline ext/opencensus_trace.c:68]: We are not releasing the memory allocated for the baggage. We can release memory in trace shutdown hook 

eg:
```
void opencensus_trace_rshutdown()
{
    /* cleanup baggage */
    zend_hash_destroy(OPENCENSUS_G(baggage));
    FREE_HASHTABLE(OPENCENSUS_G(baggage));
}
```

**Review by subintp — CHANGES_REQUESTED** [review]

File: ext/opencensus_trace.c (hunk 3 above)
Comment (subintp) [inline ext/opencensus_trace.c:68]: nit:  Define baggage length as numeric constant. Will increase readability 

`#define MAX_BAGGAGE_LENGTH 16`

File: ext/opencensus.c [hunk 4]
```diff
@@ -46,6 +46,7 @@ ZEND_END_ARG_INFO()
 ZEND_BEGIN_ARG_INFO_EX(arginfo_opencensus_trace_set_context, 0, 0, 1)
     ZEND_ARG_TYPE_INFO(0, traceId, IS_STRING, 0)
     ZEND_ARG_TYPE_INFO(0, parentSpanId, IS_STRING, 1)
+    ZEND_ARG_ARRAY_INFO(0, baggage, 1)
```
Comment (subintp) [inline ext/opencensus.c:49]: Why pass by value?

File: ext/opencensus_trace.c (hunk 3 above)
Comment (akashsharma95) [reply ext/opencensus_trace.c:68]: this is just initial length. hashtable will grow based on allocation

File: ext/opencensus.c (hunk 4 above)
Comment (akashsharma95) [reply ext/opencensus.c:49]: then you have to deref it and do other things since you are taking reference as zval from calling php script.

**Review by subintp — APPROVED** [review]

**Pull request merged**